vscode/src/chat/chat-view/chat-helpers.ts

```diff
@@ -123,3 +133,99 @@ export function getChatPanelTitle(lastDisplayText?: string, truncateTitle = true
 }
+
+/**
+ * Provides and signals updates to the current codebase identifiers to use in the chat panel.
+ */
+export class CodebaseStatusProvider implements vscode.Disposable, ContextStatusProvider {
+    private disposables: vscode.Disposable[] = []
+    private _currentCodebase: CodebaseIdentifiers | null | undefined = undefined
+    private statusCallbacks: ((provider: ContextStatusProvider) => void)[] = []
+
+    // TODO(beyang): mix of dependency injection and direct vscode API access here
+    constructor(
+        private editor: Editor,
+        private embeddingsClient: CachedRemoteEmbeddingsClient
+    ) {
+        this.disposables.push(
+            vscode.window.onDidChangeActiveTextEditor(() => this.syncCodebase()),
+            vscode.workspace.onDidChangeWorkspaceFolders(() => this.syncCodebase())
+        )
+    }
+
+    public dispose(): void {
+        for (const d of this.disposables) {
+            d.dispose()
+        }
+        this.disposables = []
+    }
+
+    public onDidChangeStatus(callback: (provider: ContextStatusProvider) => void): Disposable {
+        this.statusCallbacks.push(callback)
+        return {
+            dispose: () => {
+                this.statusCallbacks = this.statusCallbacks.filter(cb => cb !== callback)
+            },
+        }
+    }
+    public get status(): ContextGroup[] {
+        const codebase = this.currentCodebase
+        return codebase?.remote
+            ? [
+                  {
+                      name: codebase.local,
+                      providers: [
+                          {
+                              kind: 'embeddings',
+                              type: 'remote',
+                              state: 'ready',
+                              origin: this.embeddingsClient.getEndpoint(),
+                              remoteName: codebase.remote,
+                          },
+                      ],
+                  },
+              ]
+            : []
+    }
+
+    public get currentCodebase(): CodebaseIdentifiers | null {
+        if (this._currentCodebase === undefined) {
+            this._currentCodebase = CodebaseStatusProvider.fetchCurrentCodebase(this.editor)
+        }
+        for (const s of this.statusCallbacks) {
+            s(this)
+        }
+        return this._currentCodebase
+    }
+
+    private syncCodebase(): void {
+        const newCodebase = CodebaseStatusProvider.fetchCurrentCodebase(this.editor)
+        if (newCodebase === this._currentCodebase) {
+            return
+        }
+        this._currentCodebase = newCodebase
+        for (const s of this.statusCallbacks) {
+            s(this)
+        }
+    }
+
+    private static fetchCurrentCodebase(editor: Editor): CodebaseIdentifiers | null {
+        const workspaceRoot = editor.getWorkspaceRootUri()
+        if (!workspaceRoot) {
+            return null
+        }
+        const remoteUrl = repositoryRemoteUrl(workspaceRoot)
```

Comment:
```suggestion
        const remoteUrl = repositoryRemoteUrl(editor.getActiveTextEditor()?.fileUri || workspaceRoot)
```

This supports workspace with multi-repositories.

Reply:
`Editor.getWorkspaceUri` returns the workspace URI of the active editor's workspace folder.

Reply:
getWorkspaceUri doesn't change on nested repositories though, which's not necessary if that's not something we support 👍